Automattic/wp-calypso
# Jetpack Cloud: create query component to fetch JP Scan History data

#### Changes proposed in this Pull Request

- Create a Query Component to fetch the Scan History data and load it into the store.

#### Implementation notes

- This PR still uses fake data for the History section – we should address this in another PR.

#### Testing instructions

- Apply the "Jetpack Scan API: Add new /site/<blog id>/scan endpoints" patch to your sandbox (D41529-code)
- Navigate to http://calypso.localhost:3000
- Select a site with threats
- Go to the History section
- Check the JETPACK_SCAN_HISTORY_REQUEST action has been dispatched
- Check the JETPACK_SCAN_HISTORY_UPDATE action has been dispatched
- Check in Redux that the data is stored in `jetpackScan.history.data` and the request status in `jetpackScan.history.requestStatus`
- Check tests pass by running `npm run test-client:watch client/components/data/query-jetpack-scan-history/test/index.js`

See 1151678672052943-as-1170019818640873.
